libs/blog/newsletter/data-access/src/lib/state/newsletter.store.ts

```diff
@@ -0,0 +1,64 @@
+import { computed, inject } from '@angular/core';
+import { tapResponse } from '@ngrx/operators';
+import {
+  patchState,
+  signalStore,
+  withComputed,
+  withHooks,
+  withMethods,
+  withState,
+} from '@ngrx/signals';
+import { rxMethod } from '@ngrx/signals/rxjs-interop';
+import { pipe, switchMap, tap } from 'rxjs';
+
+import { SubscriberModel } from '../dto';
+import { NewsletterService } from '../infrastructure/newsletter.service';
+
+type NewsletterState = {
+  loading: 'success' | 'error' | 'init' | 'loading';
+};
+
+const initialState: NewsletterState = {
+  loading: 'init',
+};
+
+export const NewsletterStore = signalStore(
+  { providedIn: 'root' },
+  withState(initialState),
+  withComputed(({ loading }) => ({
+    loading: computed(() => loading()),
+  })),
```

Comment:
what's the point of this

Reply:
Because we shouldn't allow user to send the request again when loading is equal to `loading`. What is more, it allows us to handle with success and error state (the card displaying is depended on this computed signal)

Reply:
i think the comment was about unnecessary `withComputed` method, you should be abble to access `store.loading` without it.

Reply:
Sorry, my bad, corrected